### ISSUE TITLE
Should be flagged by avoid_shadowing_type_parameters

### ISSUE THREAD

**Issue by eernstg** [dart-lang/linter#2769]
Cf. https://github.com/dart-lang/sdk/issues/46594, the following program does not have any lints even though there are two parameters whose name is the same as a type variable declared by an enclosing entity:

```dart
class C<X> {
  C.name();
  factory C(X) => C.name(); // The parameter `X` shadows the type parameter.
  void m(int X) {} // Ditto.
}
```

The lint might be aimed at type variables shadowing type variables, but it seems useful to handle this kind of shadowing as well.

**Comment by bwilkerson** [comment]
Wouldn't these cases be handled by the enhancements mentioned in #2770? If so, I think avoid_types_as_parameter_names would be the better place to flag these cases.

**Comment by eernstg** [comment]
You're right, that's actually a better match!
